internal/app/machined/internal/phase/config/config.go

```diff
@@ -30,30 +37,95 @@ func (task *Task) TaskFunc(mode runtime.Mode) phase.TaskFunc {
 }
 
 func (task *Task) standard(r runtime.Runtime) (err error) {
+	_, err = os.Stat(constants.ConfigPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			log.Printf("did not find existing config.yaml. pulling it")
+
+			var cfgBytes []byte
+
+			cfgBytes, err = fetchConfig(r)
+			if err != nil {
+				return err
+			}
+
+			*task.cfgBytes = cfgBytes
+
+			return nil
+		}
+
+		return err
+	}
+
+	log.Printf("found an existing config.yaml. determining if we should pull anyways")
+
+	content, err := config.FromFile(constants.ConfigPath)
```

Comment:
this might be _theoretical_, but still: what if config gets corrupted/it is unreadable for some reason, or due to some bug we got it saved in a wrong way?

before this PR, as config was fetched every time, this wasn't noticeable

after this PR:

* if persist is enabled, probably there's not much we can do (if original config source is gone)
* but if persist is not enabled, it means that node won't boot anymore, while technically there's no good reason to fail the boot

so proposal:

if config can't be read, treat it as non-fatal error and fetch config, so in other words unify case above then config file doesn't exist with case below when config file can't be read/parsed or config has !.Persist. 

Basically, try reading config, if that fails, log an error, and treat that as !Persist.

Reply:
@smira reworked this to accomplish what you were talking about. if we have any issue either reading in the config from disk or parsing it, we'll force the re-fetch.

Reply:
thanks, this looks much better, but I feel we could try to unify it a bit more potentially simplifying a code.

if we had a function `readConfig() (runtime.Configurator, error)` which tries to read config, but returns an error if it's missing, we can call it, and if it returns an error or `!cfg.Persist()` try to fetch it. this would unify the code above (which checks if file exists and below which checks for readability). I think this will be less code, and we could have used this function in the sequencer as well as it seems to be doing same thing (reading config from file) over and over again?

Reply:
so my point is that the case when config doesn't exist or when it's not readable is the same case